web/client-api/src/main/java/io/deephaven/web/client/api/tree/JsTreeTable.java

```diff
@@ -725,19 +725,42 @@ private BitSet makeColumnSubscriptionBitset() {
         return columnsBitset;
     }
 
+    private void expandOrCollapseAll(double action) {
+        keyTableData = new Object[keyColumns.length + 2][1];
+        int i = keyColumns.length;
+        Js.<JsArray<Double>>cast(keyTableData[i++]).setAt(0, (double) 0);
+        Js.<JsArray<Double>>cast(keyTableData[i++]).setAt(0, action);
+        if (keyTable != null) {
+            keyTable.then(t -> {
+                t.close();
+                return null;
+            });
+            keyTable = null;
+        }
+        replaceSubscription(RebuildStep.HIERARCHICAL_TABLE_VIEW);
```

Comment:
This is common in this method and end of `setExpanded`, can refactor (as mentioned in previous comments)